metricbeat/mb/mb.go

```diff
@@ -84,6 +90,42 @@ type EventsFetcher interface {
 	Fetch() ([]common.MapStr, error)
 }
 
+// Reporter is used by a MetricSet to report events, errors, or errors with
+// metadata. The methods return false iff publishing failed because the
+// MetricSet is being closed.
+type Reporter interface {
+	Event(event common.MapStr) bool
+	ErrorWith(err error, meta common.MapStr) bool
+	Error(error) bool
```

Comment:
I think having separate methods for each of the cases (will) improve the readability of MetricSets that implement `ReportingFetcher`.

Maybe it's not clear that MetricSets don't implement this interface, they only use it. So reducing the number of methods doesn't change the amount of methods a MetricSet needs to implement (if that was your concern).